universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -233,11 +233,13 @@ public void maybeClearArray(Path path) {
 
   private void putAt(Path path, Object value) {
     checkLocked();
+    putParentIfMissing(path);
     jsonData.put(path.parentPath().toString(), path.keyName(), value);
   }
 
   private void addAt(Path path, Object value) {
     checkLocked();
+    putParentIfMissing(path);
```

Comment:
Should I add this one too? Not relevent to the checkbox thing I guess but seemed right based on concistency

Reply:
Good catch, I think that is correct. Thanks for updating!

Reply:
Hey, this looks good. I have one request though, reading the `putParentIfMissing` function, there is a desire to simplify the function and remove the recursion. These two functions are called by the `putParentIfMissing` function and calling `putParentIfMissing` here further complicates the recursion tree. Would you instead add the function call in `maybeClearArray` directly before `putAt`? I think that will work, too, and leave a less complicated codebase for anyone who aspires to remove the recursion in the future.

Reply:
Ok I did it

Reply:
Thanks!